packages/extension-dapp/README.md

```diff
@@ -35,4 +35,9 @@ api.setSigner(injector.signer);
 api.tx.balances
   .transfer('5C5555yEXUcmEJ5kkcCMvdZjUo7NGJiQJMS7vZXEeoMhj3VQ', 123456)
   .signAndSend('5DTestUPts3kjeXSTMyerHihn1uwMfLj8vU8sqF7qYrFabHE', (status) => { ... });
+
+// retrieve a RPC provider from a particular source
+// if the extension does not expose this provider, it will return an error.
+const provider = web3RpcProvider('polkadot-js', { type: 'WsProvider', payload: 'http://locahost:9944' });
```

Comment:
I went with a blacklist method: user can request any provider from extension (for now: only WsProvider, any endpoint), and background will allow/reject the request.

An alternative would be a whitelist method:
- user sends `pub(rpc.getAllProviders)`, the background will return all the Providers it can provide
- user sends `pub(rpc.startProvider)`, the background instantiates that provider (aka call `new WhateverProvider()`)